ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterAggregator.java

```diff
@@ -180,9 +357,69 @@ private StatsSnapshot reduce(StatsSnapshot statsSnapshot) {
   }
 
   /**
+   * Reduce the given {@link StatsSnapshot} whose first level mapped by PartitionClass to a shallower {@link StatsSnapshot}.
+   * by adding entries belonging to each partition together and aggregating based on partition class. The partition level
+   * would be removed after reduce work completes.
+   * <pre>
+   *             Before Reduce                   |             After Reduce
+   * --------------------------------------------------------------------------------------------
+   * {                                           |        {
+   *   value: 1000,                              |          value:1000
+   *   subMap:{                                  |          subMap:{
+   *     PartitionClass_1: {                     |            PartitionClass_1: {
+   *       value: 1000,                          |              value: 1000,
+   *       subMap: {                             |              subMap: {
+   *         Partition[1]:{                      |                Account[1]_Container[1]:{
+   *           value: 400,                       |                  value: 1000,
+   *           subMap: {                         |                  subMap: null
+   *             Account[1]_Container[1]:{       |                }
+   *               value: 400,                   |              }
+   *               subMap: null                  |            }
+   *             }                               |          }
+   *           }                                 |        }
+   *         },                                  |
+   *         Partition[2]:{                      |
+   *           value: 600,                       |
+   *           subMap: {                         |
+   *             Account[1]_Container[1]:{       |
+   *               value: 600,                   |
+   *               subMap: null                  |
+   *             }                               |
+   *           }                                 |
+   *         }                                   |
+   *       }
+   *     }
+   *   }
+   * }
+   * </pre>
+   * @param statsSnapshot the {@link StatsSnapshot} to be reduced
+   * @return the reduced {@link StatsSnapshot}
+   */
+  static StatsSnapshot reduceByPartitionClass(StatsSnapshot statsSnapshot) {
+    StatsSnapshot returnSnapshot = new StatsSnapshot(statsSnapshot.getValue(), new HashMap<>());
+    Map<String, StatsSnapshot> partitionClassSnapshots = statsSnapshot.getSubMap();
+    if (partitionClassSnapshots != null) {
```

Comment:
could you test the cases where `partitionClassSnapshots == null` and `partitionToSnapshot == null`? Codecov says these two cases are not covered.

Reply:
After revisiting the code, I find `partitionToSnapshot` won't be null and I removed the check. `partitionClassSnapshots` can only be empty. I added a unit test to verify empty `partitionClassSnapshots`.